### ISSUE TITLE
New Utils

### ISSUE THREAD

**Issue by olbrichj** [olbrichj/Futsushu#1]
## Context 🕵️‍♀️
I like the idea of `sake`. Right now I'm using `fastlane` on my projects and on projects at work. What I don't like is that I consider the whole fastlane ecosystem fragile so I hope that `sake` could be a bit more stable.

## What 🌱
I'd like to discuss the desired/expected purpose of sake. I'd appreciate something robust to simplify my CI/CD process. 

Ordinal process with `fastlane` looks like this:
1. set correct version of Xcode using `xcversion`
2. resolve provisioning using `match`
3. resolve dependencies using `carthage` or `cocoapods`
4. (in case of beta build) add beta badges to app icon using `badge` plugin in
5. run tests using `scan`
6. build app using `gym`
7. distribute build (HockeyApp/Testflight/etc.)

## Proposal 🎉
So the idea would be to add another utils which could simplify such process. I am not sure about all the mentioned tools but right now I think that util which would simplify using `xcodebuild` might be useful.

I'm open to contributing with such tool, but you know, I don't want to start with something that is not going to be useful.

What do you think? 🙂

**Comment by olbrichj** [comment]
There are multiple options to start at:
1. setting xcversion
2. doing the most common `xcodebuild`tasks (test, build, build-for-testing, etc)
3. resolve dependencies (carthage and cocoapods should be different tasks)

any other options?

**Comment by olejnjak** [comment]
Maybe I'd skip tools for Carthage and Cocoapods as they are easy to use by the Shell utility. Right now I can't think of any others.

**Comment by olbrichj** [comment]
Agree. We can add them later. Also as soon as sake supports modules, we could package corresponding tasks to make it easier for everyone.

Let's start with the `xcodebuild` methods.

Right of the bat I'd say we need:

- build with configuration (this contains: project/workspace, target, scheme, plattform, architecture, exportFormat, exportPath, exporting the archive, provisioning profile) We will definitely need reasonable defaults here
- testing (similar to the above)

Regarding the API, I'd suggest stealing ideas at fastlane.

**Comment by olbrichj** [comment]
Okay I had a look and at the moment everything needs to be within the SakeFile.swift but it's being worked on.

**Comment by olejnjak** [comment]
Yep definitely, I think inspiring in `gym` would be a good starting point.

**Comment by olbrichj** [comment]
Here is a first draft: https://github.com/olbrichj/Futsushu/blob/xcodebuild/Sakefile.swift
Feel free to change stuff.

**Comment by olbrichj** [comment]
we need multiple commands. What is each minimal subset?

Here are some examples for building and exporting an archive
http://subathrathanabalan.com/2016/01/07/building-ipa-export-archive/
